### PR TITLE
Modified Fragment creation for mutect2 to not fail for supplementary reads

### PR DESCRIPTION
**Pull request by KevinCLydon**
Modified the fragment creation code to not throw an exception in the case that there are more than 2 reads supporting a fragment and they are all supplementary, secondary, and/or duplicates.  Instead, it will just pick one of those reads to use.

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/utils/read/Fragment.java [hunk 1]
```diff
@@ -11,6 +11,7 @@
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
```
Comment (davidbenjamin) [inline src/main/java/org/broadinstitute/hellbender/utils/read/Fragment.java]: Is this new import used?

Comment (KevinCLydon) [reply src/main/java/org/broadinstitute/hellbender/utils/read/Fragment.java]: Oops, I forgot to remove that before committing.  I'll take that out

**Comment by pieterlukasse** [comment]
hi @KevinCLydon , thanks for this fix. I am preparing a custom version of 4.1.4.1 with just this patch on top of it. One question I had is: is this a safe patch to have on its own, or does it need to go in tandem with any other patch to form a proper release (and not break anything else)? Given this fix is now ~7 months old, I'm hoping insiders can maybe tell me if any other patch ended up being needed after this one, either to deal with the same issue, or to deal with any new issues introduced by it (if any).

**Comment by fleharty** [comment]
@pieterlukasse I don't know if this is an appropriate solution.  
If you are willing to filter supllementary reads, you can use this argument and it might do what you need.
--read-filter NotSupplementaryAlignmentReadFilter

**Comment by pieterlukasse** [comment]
thanks @fleharty . I can confirm that using `--read-filter NotSupplementaryAlignmentReadFilter` in my case does result in a successful run. Thanks for letting me know about this alternative. 👍

**Comment by pieterlukasse** [comment]
@fleharty I looked up other tickets for this extra --read-filter and found this https://github.com/broadinstitute/gatk/issues/6230#issuecomment-556466672 . This comment and following comments lead me to think there might be an issue with this --read-filter that was only solved in 4.1.5.0? Or has --read-filter issue also been solved in 4.1.4.1?

**Comment by pieterlukasse** [comment]
I think I understand it now: https://github.com/broadinstitute/gatk/pull/6240 did fix the `--read-filter NotSupplementaryAlignmentReadFilter` issue and got into 4.1.4.1. Not using the --read-filter does now cause another issue in 4.1.4.1 which this PR (6327) solved. Correct?

**Comment by davidbenjamin** [comment]
@pieterlukasse Almost correct: #6240 partially solved the issue but left an edge case that #6327 fixed.  The 4.1.5.0 release is the only one guaranteed to work, though 4.1.4.1 works for the great majority of samples.

**Pull request merged**